### ISSUE TITLE
Task is reported as created twice during configuration cache store run

### ISSUE THREAD

**Issue by Vampire** [gradle/gradle#24494]
If you run the `help` task with configuration cache enabled then both the task statistics from `org.gradle.internal.tasks.stats` and also a build scan show on first run (that stores CC entry) that the `help` task is created once via old API / immediately and once via new API / lazily.

If you run the second time (when CC entry is reused) then it is only displayed once but as created via old API / immediately.

While this is probably technically correct, both facts are confusing.
Maybe there could be a new category in both reports "restored from configuration cache" and maybe in the initial run even suppressed from the report?

I observed this behavior in Gradle 8.0.2.
I'd guess the twice report is since 8.0 since the first build already does parallel execution and the other quirk is probably there before already.

**Comment by ov7a** [comment]
Thank you for providing build scans.

The issue is in the backlog of the relevant team but this area of Gradle is currently not a focus one so it might take a while before a fix is made.

**Comment by wolfs** [comment]
Yes, you are correct that build scans do show the tasks loaded from the configuration cache as created eagerly. We plan to address this in one of the upcoming Gradle releases.

**Comment by wolfs** [comment]
In Gradle Enterprise 2023.2 tasks created while loading from the cache won't show up in build scans any more: 

![Image](https://github.com/gradle/gradle/assets/423186/77ff9aa9-ea3b-4787-bd21-6b155e4c4080)

When only loading, the whole `total tasks` section will not show up at all:


![Image](https://github.com/gradle/gradle/assets/423186/cf9034e6-06d0-4a1c-b04b-ff2a35f93933)


That should at least remove some of the confusion in the short term.

**Comment by wolfs** [comment]
How important is fixing `org.gradle.internal.tasks.stats`?

**Comment by Vampire** [comment]
To me? More than not at least. :-)
Especially, as build scans are not always enough, or even not available.
